### PR TITLE
[platform/cel]: Backport reboot cause API to 201811 branch

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- Why I did it**
- Add reboot cause API to support process-reboot-cause.service

**- How I did it**
- Implement chassis.get_reboot_cause platform API

**- How to verify it**
-  verify reboot cause on `/host/reboot-cause/previous-reboot-cause.txt` 
- [201811-previous-reboot-cause.log](https://github.com/Azure/sonic-buildimage/files/4661841/201811-previous-reboot-cause.log)

### CODE REVIEW

**Comment by mudsut4ke** [comment]
retest broadcom please

**Comment by lguohan** [comment]
retest mellanox please

**Comment by lguohan** [comment]
retest vsimage please

**Comment by lguohan** [comment]
can you fix the lgtm alerts?

**Comment by mudsut4ke** [comment]
> can you fix the lgtm alerts?

done

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/__init__.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import chassis
+import platform
```
Comment (jleveque) [inline device/celestica/x86_64-cel_seastone-r0/sonic_platform/__init__.py:2]: Add newline at EOF

File: device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        prev_reboot_cause = {
+            '0x11': (self.REBOOT_CAUSE_POWER_LOSS, 'Power on reset'),
+            '0x22': (self.REBOOT_CAUSE_WATCHDOG, 'Watchdog reset'),
+            '0x33': (self.REBOOT_CAUSE_HARDWARE_OTHER, 'Power cycle reset triggered by CPU')
+        }.get(hw_reboot_cause, (self.REBOOT_CAUSE_HARDWARE_OTHER, 'Unknown reason'))
+
+        if sw_reboot_cause != 'Unknown' and ( hw_reboot_cause == '0x11' or hw_reboot_cause == '0x33'):
+            prev_reboot_cause = (self.REBOOT_CAUSE_NON_HARDWARE, sw_reboot_cause)
+
+        return prev_reboot_cause
```
Comment (jleveque) [inline device/celestica/x86_64-cel_seastone-r0/sonic_platform/chassis.py:71]: Add newline at EOF

File: platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-dx010.install [hunk 3]
```diff
@@ -1,3 +1,5 @@
 dx010/scripts/dx010_check_qsfp.sh usr/local/bin
 dx010/cfg/dx010-modules.conf etc/modules-load.d
 dx010/systemd/platform-modules-dx010.service lib/systemd/system
+dx010/modules/sonic_platform-1.0-py2-none-any.whl usr/share/sonic/device/x86_64-cel_seastone-r0
+services/platform_api/platform_api_mgnt.sh usr/local/bin
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-cel/debian/platform-modules-dx010.install:5]: Add newline at EOF

File: platform/broadcom/sonic-platform-modules-cel/dx010/modules/dx010_cpld.c [hunk 4]
```diff
@@ -558,4 +637,4 @@ module_exit(cel_dx010_lpc_exit);
 MODULE_AUTHOR("Abhisit Sangjan  <asang@celestica.com>");
 MODULE_AUTHOR("Pariwat Leamsumran  <pleamsum@celestica.com>");
 MODULE_DESCRIPTION("Celestica SeaStone DX010 LPC Driver");
-MODULE_LICENSE("GPL");
+MODULE_LICENSE("GPL");
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-cel/dx010/modules/dx010_cpld.c:640]: Add newline at EOF

**Comment by mudsut4ke** [comment]
@jleveque , done in https://github.com/Azure/sonic-buildimage/pull/4619/commits/afd5faad071f7ccfaf1745b2d49297637daeff47

**Comment by mudsut4ke** [comment]
retest broadcom please

**Pull request merged**